### PR TITLE
Add support for Amazon Linux 2023 in Vulnerability Detector

### PR DESCRIPTION
**Pull request by Leoquicenoz**
|Related issue|
|---|
|[#18526](https://github.com/wazuh/wazuh/issues/18526)|

## Description

The purpose of this PR is to add support for Amazon Linux 2023 in the vulnerability detector module.

First, we generate our own feed in `JSON` format and upload it to the following links (from where we download it): 
[Amazon Linux 2023 feed](https://feed.wazuh.com/vulnerability-detector/ALAS/2023/alas2023.json.gz) and [Amazon Linux 2023 metadata](https://feed.wazuh.com/vulnerability-detector/ALAS/2023/alas2023.meta)

The vulnerabilities are then parsed and inserted into the database.

And finally, when an Amazon Linux 2023 agent is detected, then if the configuration is well set, it scans the agent and reports the vulnerabilities it contains.

## Configuration options

The Vulnerability Detector configuration has been extended to support the added OS (AL2023):
```xml
<vulnerability-detector>
    ...
    <!-- Amazon Linux OS vulnerabilities -->
    <provider name="alas">
      <enabled>yes</enabled>
      <os>amazon-linux</os>
      <os>amazon-linux-2</os>
      <os>amazon-linux-2022</os>
      <os>amazon-linux-2023</os>
      <update_interval>1h</update_interval>
    </provider>
    ...
</vulnerability-detector>
```

## Logs/Alerts example

Here we can see the logs when fetching the ALAS2023 feed:

```
2023/08/24 16:28:50 wazuh-modulesd:vulnerability-detector[221457] wm_vuln_detector.c:5330 at wm_vuldet_check_feed(): INFO: (5400): Starting 'Amazon Linux 2023' database update.
2023/08/24 16:28:50 wazuh-modulesd:vulnerability-detector[221457] wm_vuln_detector.c:5228 at wm_vuldet_fetch_feed(): DEBUG: (5484): Cleaning metadata for target 'Amazon-Linux-2023'
2023/08/24 16:28:50 wazuh-modulesd:vulnerability-detector[221457] wm_vuln_detector.c:5244 at wm_vuldet_fetch_feed(): DEBUG: (5403): Fetching feed from '/home/user/alas2023.json.gz'
2023/08/24 16:28:50 wazuh-modulesd:vulnerability-detector[221457] wm_vuln_detector.c:8244 at wm_vuldet_index_json(): DEBUG: (5408): Updating from '/home/user/alas2023.json.gz'
2023/08/24 16:28:50 wazuh-modulesd[221457] file_op.c:3392 at w_uncompress_bz2_gz_file(): DEBUG: The file '/home/user/alas2023.json.gz' was successfully uncompressed into 'tmp/vuln-temp-fitted'
2023/08/24 16:28:50 wazuh-modulesd:vulnerability-detector[221457] wm_vuln_detector.c:4970 at wm_vuldet_index_feed(): DEBUG: (5414): Refreshing 'Amazon Linux 2023' databases.
2023/08/24 16:28:50 wazuh-modulesd:vulnerability-detector[221457] wm_vuln_detector.c:3283 at wm_vuldet_insert(): DEBUG: (5415): Inserting vulnerabilities.
2023/08/24 16:28:50 wazuh-modulesd:vulnerability-detector[221457] wm_vuln_detector.c:3312 at wm_vuldet_insert(): DEBUG: (5419): Inserting ALAS vulnerabilities section.
2023/08/24 16:28:50 wazuh-modulesd:vulnerability-detector[221457] wm_vuln_detector.c:4976 at wm_vuldet_index_feed(): DEBUG: (5427): Refresh of 'Amazon Linux 2023' database finished.
2023/08/24 16:28:50 wazuh-modulesd:vulnerability-detector[221457] wm_vuln_detector.c:4982 at wm_vuldet_index_feed(): DEBUG: remove(tmp/vuln-temp): No such file or directory
2023/08/24 16:28:50 wazuh-modulesd:vulnerability-detector[221457] wm_vuln_detector.c:4988 at wm_vuldet_index_feed(): DEBUG: remove(tmp/vuln-temp.bz2): No such file or directory
2023/08/24 16:28:50 wazuh-modulesd:vulnerability-detector[221457] wm_vuln_detector.c:5353 at wm_vuldet_check_feed(): INFO: (5430): The update of the 'Amazon Linux 2023' feed finished successfully.

```

Vulnerability log displayed during Amazon Linux 2023 packages scan:

```
2023/09/06 10:21:07 wazuh-modulesd:vulnerability-detector[11838] wm_vuln_detector.c:1579 at wm_vuldet_process_agent_vulnerabilities(): DEBUG: (5466): Sending vulnerabilities report for agent '001'
2023/09/06 10:21:07 wazuh-modulesd:vulnerability-detector[17501] wm_vuln_detector.c:1686 at wm_vuldet_process_agent_vulnerabilities(): DEBUG: (5468): The 'vim-common' package (2:9.0.1160-1.amzn2023.0.2) from agent '001' is vulnerable to 'CVE-2022-47024'. Condition: 'Package less than 9.0.1314-1.amzn2023.0.2'
2023/09/06 10:21:07 wazuh-modulesd:vulnerability-detector[17501] wm_vuln_detector.c:1686 at wm_vuldet_process_agent_vulnerabilities(): DEBUG: (5468): The 'vim-data' package (2:9.0.1160-1.amzn2023.0.2) from agent '001' is vulnerable to 'CVE-2022-47024'. Condition: 'Package less than 9.0.1314-1.amzn2023.0.2'
2023/09/06 10:21:08 wazuh-modulesd:vulnerability-detector[17501] wm_vuln_detector.c:1686 at wm_vuldet_process_agent_vulnerabilities(): DEBUG: (5468): The 'vim-enhanced' package (2:9.0.1160-1.amzn2023.0.2) from agent '001' is vulnerable to 'CVE-2022-3324'. Condition: 'Package less than 9.0.1314-1.amzn2023.0.2'
2023/09/06 10:21:10 wazuh-modulesd:vulnerability-detector[17501] wm_vuln_detector.c:1717 at wm_vuldet_process_agent_vulnerabilities(): DEBUG: (5482): A total of '0' vulnerabilities have been reported for agent '001' thanks to the 'NVD' feed.
2023/09/06 10:21:10 wazuh-modulesd:vulnerability-detector[17501] wm_vuln_detector.c:1718 at wm_vuldet_process_agent_vulnerabilities(): DEBUG: (5482): A total of '66' vulnerabilities have been reported for agent '001' thanks to the 'vendor' feed.
2023/09/06 10:21:10 wazuh-modulesd:vulnerability-detector[17501] wm_vuln_detector.c:1720 at wm_vuldet_process_agent_vulnerabilities(): DEBUG: (5469): A total of '66' vulnerabilities have been reported for agent '001'
```

## Tests

Minimum checks required 
- Compilation without warnings in every supported platform
  - [x] Linux
- [x] Source installation
- [ ] Package installation
- [ ] Source upgrade
- [ ] Package upgrade
- [ ] Review logs syntax and correct language
- [ ] QA templates contemplate the added capabilities

Depending on the affected OS 
- Memory tests for Linux
  - [x] Scan-build report
  - [ ] Coverity
  - [x] Valgrind (memcheck and descriptor leaks check)
  - [ ] AddressSanitizer

Checks for huge PRs that affect the product more generally 
- [x] Retrocompatibility with older Wazuh versions
- [x] Working on cluster environments
- [x] Configuration on demand reports new parameters
- [x] The data flow works as expected (agent-manager-api-app)
- [x] Added unit tests (for new features)
- [ ] Stress test for affected components

### CODE REVIEW

File: src/unit_tests/wazuh_modules/vulnerability_detector/test_wm_vuln_detector.c [hunk 1]
```diff
@@ -2541,9 +2541,12 @@ void test_wm_vuldet_generate_os_and_kernel_package_amazon_linux(void **state)
 {
     scan_agent *agent = *state;
     sqlite3 *db = (sqlite3 *)1;
-    int dist_ver_ref[] = {FEED_ALAS1, FEED_ALAS2, FEED_ALAS2022};
-    char *alas_os_release[] = {"AMI", "2", "2022"};
-    char *alas_os_package[] = {"linux", "linux_2", "linux_2022"};
+    int dist_ver_ref[] = {FEED_ALAS1, FEED_ALAS2, FEED_ALAS2022, FEED_ALAS2023};
+    // char *alas_os_release[] = {"AMI", "2", "2023"};
+    // char *alas_os_package[] = {"linux", "linux_2", "linux_2023"};
+
```
Comment (mateocervilla) [inline src/unit_tests/wazuh_modules/vulnerability_detector/test_wm_vuln_detector.c:2547]:
```suggestion
```

File: src/unit_tests/wazuh_modules/vulnerability_detector/test_wm_vuln_detector.c [hunk 2]
```diff
@@ -2541,9 +2541,12 @@ void test_wm_vuldet_generate_os_and_kernel_package_amazon_linux(void **state)
 {
     scan_agent *agent = *state;
     sqlite3 *db = (sqlite3 *)1;
-    int dist_ver_ref[] = {FEED_ALAS1, FEED_ALAS2, FEED_ALAS2022};
-    char *alas_os_release[] = {"AMI", "2", "2022"};
-    char *alas_os_package[] = {"linux", "linux_2", "linux_2022"};
+    int dist_ver_ref[] = {FEED_ALAS1, FEED_ALAS2, FEED_ALAS2022, FEED_ALAS2023};
```
Comment (mateocervilla) [inline src/unit_tests/wazuh_modules/vulnerability_detector/test_wm_vuln_detector.c:2544]:
```suggestion
    int dist_ver_ref[] = {FEED_ALAS1, FEED_ALAS2, FEED_ALAS2022, FEED_ALAfgjfgj23};
```

File: src/unit_tests/wazuh_modules/vulnerability_detector/test_wm_vuln_detector.c [hunk 3]
```diff
@@ -2597,6 +2603,7 @@ void test_wm_vuldet_generate_os_and_kernel_package_amazon_linux(void **state)
             expect_sqlite3_step_call(SQLITE_DONE);
         }
 
+
```
Comment (mateocervilla) [inline src/unit_tests/wazuh_modules/vulnerability_detector/test_wm_vuln_detector.c:2606]:
```suggestion
```

File: src/unit_tests/wazuh_modules/vulnerability_detector/test_wm_vuln_detector.c [hunk 4]
```diff
@@ -2617,7 +2624,8 @@ void test_wm_vuldet_generate_os_and_kernel_package_amazon_linux(void **state)
         expect_string(__wrap_sqlite3_bind_text, buffer, "94b6f7b3c1d905aae22a652448df6372da98e5b8");
         expect_value(__wrap_sqlite3_bind_text, pos, 12);
         expect_string(__wrap_sqlite3_bind_text, buffer, "OS");
-
+       
+
```
Comment (mateocervilla) [inline src/unit_tests/wazuh_modules/vulnerability_detector/test_wm_vuln_detector.c:2628]:
```suggestion
```

**Review by MarcelKemp — CHANGES_REQUESTED** [review]
We will avoid adding the changelog line, as this will be done in a separate issue.

**Review by MarcelKemp — APPROVED** [review]

**Pull request merged**